src/internal/m365/collection/site/backup.go

```diff
@@ -185,17 +188,33 @@ func CollectLists(
 			el.AddRecoverable(ctx, clues.WrapWC(ctx, err, "creating list collection path"))
 		}
 
-		collection := NewPrefetchCollection(
-			bh,
-			dir,
-			ac,
-			scope,
-			su,
-			bpc.Options)
-
-		collection.AddItem(
-			ptr.Val(list.GetId()),
-			ptr.Val(lmt))
+		if lmt.IsZero() {
```

Comment:
This may cause NPE if `lmt` is nil. You can use `ptr.Val(lmt).IsZero()`

Reply:
Also can you please explain the logic behind why we are using zero mod time to select between prefetch and lazy collections?

Reply:
@pandeyabs I believe kopia uses `modtime` to determine if it needs to read an item or not. So in case if we get an invalid modtime from graph, the idea is to use a prefetch coll for that. Probably prefetch will never be invoked its just for precautinary purpose.

Reply:
Thanks, makes sense.